codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/core/KotlinSymbolProvider.kt

```diff
@@ -253,6 +284,13 @@ class KotlinSymbolProvider(private val model: Model, private val settings: Kotli
         return builder
     }
 
+    private fun getDefaultValueForNumber(shape: Shape, value: String) = when (shape) {
+        is LongShape -> "${value}L"
+        is FloatShape -> if (value.matches("[0-9]*\\.[0-9]+".toRegex())) "${value}f" else "$value.0f"
```

Comment:
**Question**: Is the decimal check necessary? The `f` type specifier should mean that a decimal component is not required (i.e., `5f` and `5.0f` are identical).